Keen/Event.cs

```diff
@@ -53,60 +91,63 @@ internal class Event : IEvent
         /// <returns></returns>
         public async Task<IEnumerable<CachedEvent>> AddEvents(JObject events)
         {
+            if (string.IsNullOrWhiteSpace(_writeKey))
+            {
+                throw new KeenException("An API WriteKey is required to add events.");
+            }
+
             var content = events.ToString();
 
-            using (var client = new HttpClient())
-            using (var contentStream = new StreamContent(new MemoryStream(Encoding.UTF8.GetBytes(content))))
+            var responseMsg = await _keenHttpClient
+                .PostAsync(_eventsRelativeUrl, _writeKey, content)
+                .ConfigureAwait(continueOnCapturedContext: false);
+            var responseString = await responseMsg
+                .Content
+                .ReadAsStringAsync()
+                .ConfigureAwait(continueOnCapturedContext: false);
+
+            JObject jsonResponse = null;
+
+            try
+            {
+                // Normally the response content should be parsable JSON,
+                // but if the server returned a 404 error page or something
+                // like that, this will throw. 
+                jsonResponse = JObject.Parse(responseString);
+
+                // TODO : Why do we not call KeenUtil.CheckApiErrorCode(jsonResponse); ??
```

Comment:
Not understanding these small discrepancies was the only reason I didn't push all the error handling stuff down into KeenHttpClient while I was at it. I might take another stab at running through the shape of responses in error conditions to see if we can just consolidate this and push it down a layer. Really the code in Event/EventCollection/Queries should get trimmed down quite a bit, and be mostly concerned with generating payload and parsing responses, and tons of *that* code is currently copy/paste code too, so hopefully after another two rounds of refactor, I'm thinking we can be at about half the lines of code in those three classes, maybe less.